### PR TITLE
feat(lint): add `no-head-element` from `eslint-plugin-next`

### PR DESCRIPTION
**Pull request by kaioduarte**
## Summary
Implement [no-head-element](https://nextjs.org/docs/messages/no-head-element) from `eslint-plugin-next`.

I'm willing to port the other rules if they're relevant to the project.

## Test Plan
Snapshots + CI

### CODE REVIEW

File: crates/biome_js_analyze/src/lint/nursery/no_next_head_element.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let element = ctx.query();
+        let name = element.name().ok()?.name_value_token()?;
+
+        if name.text_trimmed() == "head" {
+            let path = ctx.file_path().as_os_str().to_str()?;
+            let is_in_app_dir = path.contains(&format!("app{}", std::path::MAIN_SEPARATOR));
```
Comment (ematipico) [inline crates/biome_js_analyze/src/lint/nursery/no_next_head_element.rs:67]: The paths provided are always Unix-like paths, so there's no need to check against `MAIN_SEPARATOR`, which changes based on the OS. This means that the code won't work on Windows.

Comment (ematipico) [inline crates/biome_js_analyze/src/lint/nursery/no_next_head_element.rs:67]: Line 67 allocates a string every time, which is not ideal. I suggest an alternative: use [`path.ancestors`](https://doc.rust-lang.org/std/path/struct.Path.html#method.ancestors), then called `rev()` to start from the end of the iterator, and the first `app` that is a folder you find will trigger the rule.

File: crates/biome_js_analyze/src/lint/nursery/no_next_head_element.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        None
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, range: &Self::State) -> Option<RuleDiagnostic> {
+        return Some(RuleDiagnostic::new(
+            rule_category!(),
+            range,
+            markup! {
+                "Do not use "<Emphasis>"<head>"</Emphasis>" element. Use "<Emphasis>"<Head />"</Emphasis>" from "<Emphasis>"next/head"</Emphasis>" instead."
+            },
```
Comment (ematipico) [inline crates/biome_js_analyze/src/lint/nursery/no_next_head_element.rs:83]: Please refer to the contribution guide, the [three rule pillars](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md#explain-a-rule-to-the-user). This diagnostic doesn't explain why the use of `<head>` is an error.

File: crates/biome_test_utils/src/lib.rs [hunk 3]
```diff
@@ -204,13 +204,25 @@ pub fn code_fix_to_string<L: ServiceLanguage>(source: &str, action: AnalyzerActi
...
+        }
+
+        rule_name = group_name;
+        group_name = component.to_str().unwrap_or_default();
+    }
 
-    let group_folder = rule_folder.parent().unwrap();
-    let group_name = group_folder.file_name().unwrap();
+    if !root_found {
+        panic!("Failed to find group and rule");
```
Comment (ematipico) [inline crates/biome_test_utils/src/lib.rs:222]: Can you provide a better message? This is mostly for us developers, so if this panics, we should provide a message that tells the developer how to fix the issue.

File: crates/biome_js_analyze/src/lint/nursery/no_next_head_element.rs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    ///     <Head>
+    ///       <title>All good!</title>
+    ///     </Head>
+    ///   )
+    /// }
+    /// ```
+    pub NoNextHeadElement {
+        version: "next",
+        name: "noNextHeadElement",
+        language: "js",
```
Comment (ematipico) [inline crates/biome_js_analyze/src/lint/nursery/no_next_head_element.rs:48]:
```suggestion
        language: "jsx",
```

File: crates/biome_js_analyze/src/lint/nursery/no_next_head_element.rs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    ///   return (
+    ///     <Head>
+    ///       <title>All good!</title>
+    ///     </Head>
+    ///   )
+    /// }
+    /// ```
+    pub NoNextHeadElement {
+        version: "next",
+        name: "noNextHeadElement",
```
Comment (ematipico) [inline crates/biome_js_analyze/src/lint/nursery/no_next_head_element.rs:47]: The rule should be called `noHeadElement`. The documentation will explain to the users that the rule is meant for a Next.js project

**Review by ematipico — CHANGES_REQUESTED** [review]
Some important changes to address:
- the name of the rule
- the documentation
- the diagnostic

File: crates/biome_js_analyze/src/lint/nursery/no_next_head_element.rs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    ///
+    /// A `<head>` element was used to include page-level metadata, but this can
+    /// cause unexpected behavior in a Next.js application. Use Next.js' built-in
+    /// `next/head` component instead.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    /// ```jsx,expect_diagnostic
+    /// // /pages/index.jsx
```
[the remark this answers is not in the document]
Comment (kaioduarte) [reply crates/biome_js_analyze/src/lint/nursery/no_next_head_element.rs:21]: It actually runs everywhere except inside `/app` folder. I'll change that comment.

File: crates/biome_js_analyze/src/lint/nursery/no_head_element.rs [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        None
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, range: &Self::State) -> Option<RuleDiagnostic> {
+        return Some(RuleDiagnostic::new(
+            rule_category!(),
+            range,
+            markup! { "Don't use "<Emphasis>"<head>"</Emphasis>" element." },
+        ).note(markup! {
+            "Using "<Emphasis>"<head>"</Emphasis>" element can cause unexpected behavior in a Next.js application. Use "<Emphasis>"<Head />"</Emphasis>" from "<Emphasis>"next/head"</Emphasis>" instead."
```
Comment (ematipico) [inline crates/biome_js_analyze/src/lint/nursery/no_head_element.rs:86]:
```suggestion
            "Using the "<Emphasis>"<head>"</Emphasis>" element can cause unexpected behavior in a Next.js application. Use "<Emphasis>"<Head />"</Emphasis>" from "<Emphasis>"next/head"</Emphasis>" instead."
```

File: crates/biome_js_analyze/src/lint/nursery/no_head_element.rs [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+declare_lint_rule! {
+    /// Prevent usage of `<head>` element in a Next.js project.
+    ///
+    /// Next.js provides a specialized `<Head />` component from `next/head` that manages
+    /// the `<head>` tag for optimal server-side rendering, client-side navigation, and
+    /// automatic deduplication of tags such as `<meta>` and `<title>`.
+    ///
+    /// This rule only checks outside of the `app/` directory, as it's typically
+    /// handled differently in Next.js.
```
Comment (ematipico) [inline crates/biome_js_analyze/src/lint/nursery/no_head_element.rs:18]:
```suggestion
    /// This rule only checks files that are outside of the [`app/` directory](https://nextjs.org/docs/app), as it's typically
    /// handled differently in Next.js.
```

**Review by ematipico — APPROVED** [review]

**Review by togami2864 — APPROVED** [review]

**Pull request merged**